packages/gen-typescript-declarations/src/gen-ts.ts

```diff
@@ -463,17 +536,43 @@ class TypeGenerator {
       ],
       returns: new ts.IntersectionType([
         new ts.NameType('T'),
-        new ts.NameType(mixinName + 'Constructor'),
-        ...Array.from(this.transitiveMixins(feature))
-            .map((mixin) => new ts.NameType(mixin + 'Constructor'))
+        new ts.NameType(constructorName),
+        ...[...transitiveMixins].map(
+            (mixin) =>
+                new ts.NameType([...mixin.identifiers][0] + 'Constructor'))
```

Comment:
nit: I like putting expressions like `[...mixin.identifiers][0] + 'Constructor')` into a named variable or function to make it more self documenting. Since you use this below too, it'll document both cases. And would it be used for `constructorName` as well? (why is one using `feature.name` and the other the first identifier?)

Reply:
I just forgot that `.name` was a thing, switched to that.